torchgen/native_function_generation.py

```diff
@@ -303,7 +303,7 @@ def generate_function(
         func.name.unambiguous_name()
         if func.kind() == SchemaKind.out
         else cpp.name(func)
-    )
+    ) + "_symint"  # TODO: hmm
```

Comment:
why is _symint assumed here? shouldn't this be driven by the yaml?

Reply:
Sooo I am not entirely sure here, but the idea is that these kernels are autogenerated, and they handle SymInts, so we should give them symint names so they get the right signature. It's a bit dodgy though... I got here by just tweaking until it compiled

Reply:
that sounds reasonable

clarification: these 'generated' kernels are going to call into handwritten kernels (adapting from functional to out or vice versa) right?  if so, would we be potentially having a situation like this?
foo_out <-- handwritten, no symint support
foo_symint <-- generated, just trivially unwraps symints and expects ints, calls foo_out

if so, i still don't think that's a problem.  other than, we should probably make it clearer what's going on

Reply:
Err, not really? The generated kernels typically call into other API as defined by the codegen; e.g., a generated out will generate a call to functional and then copy_ (but the functional call is not a direct native:: call; it'll just be to some dispatched api).